src/components/TagItem.vue

```diff
@@ -0,0 +1,224 @@
+<template>
+	<div class="tag-group">
+		<button class="tag-group__label"
+			:style="{
+				color: convertHex(tag.color, 1),
+				'background-color': convertHex(tag.color, 0.15)
+			}">
+			{{ tag.displayName }}
+		</button>
+		<Actions :force-menu="true">
+			<NcAppNavigationIconBullet v-if="editing"
+				slot="icon"
+				:color="tag.color" />
+			<NcActionButton v-if="renameTagLabel"
+				@click="openEditTag">
+				<template #icon>
+					<IconRename :size="17" />
+				</template>
+				{{ t('mail','Edit name and color') }}
+			</NcActionButton>
+			<ActionInput v-if="renameTagInput"
+				:value="tag.displayName"
+				@submit="renameTag(tag, $event)" />
+			<ActionText
+				v-if="showSaving">
+				<template #icon>
+					<IconLoading :size="20" />
+				</template>
+				{{ t('mail', 'Saving new tag name …') }}
+			</ActionText>
+
+			<NcColorPicker class="app-navigation-entry-bullet-wrapper" :value="`#${tag.color}`" @input="updateColor">
+				<div :style="{ backgroundColor: getColor }" class="color0 icon-colorpicker app-navigation-entry-bullet" />
+			</NcColorPicker>
+		</Actions>
+		<NcButton v-if="!isSet(tag.imapLabel)"
+			class="tag-actions"
+			@click="addTag(tag.imapLabel)">
+			{{ t('mail','Set tag') }}
+		</NcButton>
+		<NcButton v-else
+			class="tag-actions"
+			@click="removeTag(tag.imapLabel)">
+			{{ t('mail','Unset tag') }}
+		</NcButton>
```

Comment:
@kesselb i have this warn and error:
`vue.runtime.esm.js:4605 [Vue warn]: Error in render: "TypeError: Cannot read properties of undefined (reading 'some')"

found in

---> <TagItem> at src/components/TagItem.vue`

`vue.runtime.esm.js:4605 [Vue warn]: Error in render: "TypeError: Cannot read properties of undefined (reading 'some')"

found in

---> <TagItem> at src/components/TagItem.vue`

and i think it is because of the position of this code here. Im not sure where else to add these 2 buttons.

Reply:
either this.envelopes or this.$store.getters.getEnvelopeTags(envelope.databaseId) evaluates to undefined